ios/engine/KMEI/KeymanEngine/Classes/Constants.swift

```diff
@@ -51,4 +51,25 @@ public struct Constants {
                                                           font: defaultFont,
                                                           oskFont: nil,
                                                           isCustom: false)
+
+  /// Is the process of a custom keyboard extension.
+  public static let isSystemKeyboard: Bool = {
```

Comment:
(disclaimer: I don't get how `isSystemKeyboard` gets used in iOS. During run-time, do in-app keyboards return false and system keyboards return true?)

I would expect things in Constants to be immutable.
Since `isSystemKeyboard` can return *true* or *false*, would it be better to move this to some utility file?

Reply:
Yep. In iOS, the app and system keyboard run in two separate processes. In the app's process, `isSystemKeyboard == false` and in the system keyboard `isSystemKeyboard == true`.

I was actually thinking about whether this belongs in `Constants` as well. It is immutable (due to `let`) but the value is determined at runtime. I was also iffy about `keymanBundle` but I think `sdkVersion` can stay.

Reply:
I think using the static method to return a dynamic value in this case is fine. Typically I would agree with @darcywong00 on this, but this is a rare case that I think works.